content/general/limits.textile

```diff
@@ -0,0 +1,314 @@
+
+h3(#inbound-message). Inbound message rate
+
+The inbound message rate is the maximum number of messages that can be received per second on a "realtime connection":/realtime/connection.
+
+If the inbound message rate is exceeded for a connection, a "local per second limit":#time will be applied to that connection.
+
+h3(#connection-state). Connection state recovery
+
+Connection state recovery is the duration the Ably service will keep a connection alive for if the client connection drops. This accounts for scenarios such as where a client is switching between WiFi and 4G, or entering a tunnel. If the client reestablishes its connection within the allowed duration, all messages and presence events will be replayed to that client in order.
+
+If a connection isn't reestablished in time, the client is detached from its channels and the error code @80008@ is returned.
+
+h2(#channel). Channel limits
+
+Channel limits relate to the number, rate and membership of "channels":/glossary#channels on your account.
+
+|_=. Limit |_=. Free |_=. Self-service |_=. Business |_=. Enterprise |
+| "Number of subscribers":#subscribers (per channel) |\4=. Unlimited |
+| "Presence members":#presence-members (per channel) | 50 | 200 | 200 | Custom |
+| "Channel creation rate":#channel-create (per second) | 20 | >=50 | >=50 | Custom |
+| "Message publish rate":#message-publish-rate (per second) | 15 | 50 | 50 | Custom |
+
+h3(#subscribers). Number of subscribers
+
+There is no limit on the number of clients that can be subscribed to a channel.
+
+h3(#presence-members). Presence members
+
+The number of clients that can use the "presence feature":/realtime/presence in a channel is limited as a precaution. If a large number of clients are subscribed to presence events on a channel then the amount of messages can scale very quickly.
+
+As an example, consider 200 clients subscribed to presence events on a channel and all of them join and leave the presence set within a few minutes. This would result in the following messages:
+
+* 200 presence messages published for the enter event.
+* 200 x 200 (40,000) messages subscribed to for the enter events.
+* 200 presence messages published for the leave event.
+* 200 x 200 (40,000) presence messages subscribed to for the leave event.
+
+This highlights the potential for 80,400 messages to be sent in a very short space of time on a single channel.
+
+The number of presence members is a "quantity-based limit":#quantity-rates. Any clients that attempt to join the presence set over the presence member limit will be rejected and the error code @91003@ will be returned.
+
+h3(#channel-create). Channel creation rate
+
+The channel creation rate is the number of channels that can be created per second.
+
+The limit is calculated based on the hard limit for "peak channels":#peak-channels, with a minimum value of 50 per second for self-service and business accounts.
```

Comment:
Might suggest adding a link here to the limits calculator so that customers can calculate the quota needed

Reply:
Perhaps using the [full calculator](https://ably.com/pricing/full-calculator) would be good so all the limits described here are visible in general?

Reply:
Thanks for the suggestion both. I've added a link at the top of the section rather than linking it in individual limits.